cmd/juju/common/flags.go

```diff
@@ -58,6 +58,28 @@ func (f *ConfigFlag) Set(s string) error {
 	return nil
 }
 
+// SetAttrsFromYAML sets the attributes from a slice of bytes. The bytes are
+// expected to be YAML parsable and align to the attrs type of
+// map[string]interface{}.
+// This will over write any attributes that already exist if found in the YAML
+// configuration.
+func (f *ConfigFlag) SetAttrsFromYAML(input []byte) error {
+	if len(input) == 0 {
+		return errors.NotValidf("empty yaml")
+	}
+	attrs := make(map[string]interface{})
+	if err := yaml.Unmarshal(input, &attrs); err != nil {
```

Comment:
It really sucks that the yaml package does not accept an `io.Reader` for such operations (e.g. like json.Decoder does).